training/run_eval.py

```diff
@@ -702,23 +738,55 @@ def _forward_time(*args, **kwargs):
 
     stats_dataset = DatasetDict()
 
-    all_stats = {"rtf": 0, "wer": 0}
+    all_stats = {"rtf": 0, "wer": 0, "tokens_per_sec": 0}
     rtf_stats = {
         "times_audio_total": 0,
         "times_transcription_total": 0,
     }
 
+    def benchmark_gen_time():
+        if model_pipeline is None:
+            dummy_encoder_outputs = BaseModelOutput(
+                torch.randn((data_args.batch_size, model.config.max_source_positions, model.config.d_model),
+                             dtype=model.dtype,
+                             device=model.device
+                )            
+            )
+
+            # benchmark time to generate fixed number of tokens
+            n_tokens = data_args.num_tokens
+            start_time = time.time()
+            _ = model.generate(
+                encoder_outputs=dummy_encoder_outputs,
+                min_new_tokens=n_tokens,
+                max_new_tokens=n_tokens,
+                **gen_kwargs
```

Comment:
Because you pass the `gen_kwargs` here, we'll also have the same decoder input ids as usual generation:
```
<|startoftranscript|> <|language|> <|task|> <|notimestamps|> ...
```

=> you can use the same logic as you do in `benchmark` to ignore these tokens from the generated time

Reply:
Since we are forcing generation of `n_tokens`, we already know that the total number of generated tokens will be `n_tokens * data_args.batch_size` →  no need to call `numel()` and ignore decoder input ids ! very slightly more efficient